lib/classes/Crittr.class.js

```diff
@@ -723,7 +724,13 @@ class Crittr {
                     debug('evaluateUrl - Extracting critical selectors');
                     await page.waitForTimeout(250); // Needed because puppeteer sometimes isn't able to handle quick tab openings
                     if (this.options.takeScreenshots === true) {
-                        const screenName = url.replace(/[^\w\s]/gi, '_') + '.png';
+                        let screenName = url.replace(/[^\w\s]/gi, '_') + '.png';
+                        if (typeof this.options.screenshotNameGenerator === 'function') {
+                            screenName = `${await this.options.screenshotNameGenerator(url)}.png`;
```

Comment:
For readability create name outside of literal string as a variable

Reply:
[1227a81](https://github.com/hummal/crittr/pull/38/commits/1227a8178d06f66d5219df98b027a55ad8853d2f)